arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1497,6 +1493,11 @@ auto ClusterInfo::loadPlan() -> consensus::index_t {
       continue;
     }
 
+    if (ServerState::instance()->isCoordinator()) {
+      // invalidate query plan caches for every changed database.
+      vocbase->queryPlanCache().invalidateAll();
```

Comment:
This is by database. And I think it is completely right, since this will only touch databases which are actually changed as far as I can see. Looks good.